benchmarking/data/simulated_data/mock_student_provider.py

```diff
@@ -241,3 +266,30 @@ def attribute_values_from_range(
         replace=False,
         generator=_generator,
     )
+
+
+def _generate_attribute_values_without_probability(
+    attribute_ranges: Dict[int, AttributeRangeConfig], class_size: int
+):
+    probabilistic_attributes = {}
+    for attribute_id, attribute_range_config in attribute_ranges.items():
+        if isinstance(attribute_range_config[0], (int, AttributeValueEnum)):
+            continue
+
+        # Assume if the range is probabilistic, the number of value per attribute is 0
+        if (sum([_[1] for _ in attribute_range_config])) != 1:
+            raise ValueError(f"Probabilistic attribute range must sum to 100$")
+
+        probabilistic_attributes[attribute_id] = []
+        for attribute_range in attribute_range_config:
+            probabilistic_attributes[attribute_id].extend(
+                [
+                    attribute_range[0]
```

Comment:
nit;

Feel like we should just name this variable, like `attribute_value, attribute_probability = attribute_range`